libs/librepcb/common/cam/excellongenerator.cpp

```diff
@@ -47,7 +47,14 @@ ExcellonGenerator::~ExcellonGenerator() noexcept {
 
 void ExcellonGenerator::drill(const Point&          pos,
                               const PositiveLength& dia) noexcept {
-  mDrillList.insert(*dia, pos);
+  mDrillList.insert(dia, pos);
+  if (!mToolList.contains(dia)) mToolList.append(dia);
+}
+
+void ExcellonGenerator::slot(const Point& start, const Point& end,
+                             const PositiveLength& width) noexcept {
+  mSlotList.insert(width, qMakePair(start, end));
+  if (!mToolList.contains(width)) mToolList.append(width);
 }
 
 void ExcellonGenerator::generate() {
```

Comment:
Since the tool list is no longer sorted (like `QMultiMap` which was used previously), the tool list now need to be sorted manually:

```suggestion
void ExcellonGenerator::generate() {
  std::sort(mToolList.begin(), mToolList.end());
```

I suspect this fixes the failed unit test (see CI output).